src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -5205,6 +5238,10 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 this->visit_expr(*x.m_elts[i]);
                 expr = ASRUtils::EXPR(tmp);
                 if (!ASRUtils::check_equal_type(ASRUtils::expr_type(expr), type)) {
+                	//set the tmp to use it in the error message.(copied from the end of this function)              	
+        			ASR::ttype_t* list_type = ASRUtils::TYPE(ASR::make_List_t(al, x.base.base.loc, type));
+        			tmp = ASR::make_ListConstant_t(al, x.base.base.loc, list.p,
+            			list.size(), list_type);
                     throw SemanticError("All List elements must be of the same type for now",
```

Comment:
I have the following idea in mind, check if it's possible to implement or handle. If any blockers, do let me know.

Instead of throw, do `diag.add(diag::Diagnostic(...))` and return. Check if diag has some error and throw error based on that.

Reply:
Okay this is doable,but I need to know what benefit we'll get after replacing a throw statement with `diag.add()`,so I can reason my fix as I don't quite get what you mean .

I think I could handle the error directly without creating ASR element. If that what you imply, that would make the code change in the `visit_assign()` only,  and (`visit_set()` , `visit_dict()` , `visit_list()` ) would remain the same.